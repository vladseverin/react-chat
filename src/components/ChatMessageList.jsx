import React from 'react';
import { withStyles } from 'material-ui/styles';
import ChatMessage from './ChatMessage.jsx';
import History from './History.jsx';

const styles = theme => ({
  messagesWrapper: {
    overflowX: 'hidden',
    width: '100%',
    height: '100%',
    paddingTop: theme.spacing.unit * 3,
    paddingBottom: '120px',
  }
});

class CHatMessageList extends React.Component {
  componentDidMount() {
    this.scrollDownHistory();
  }

  componentDidUpdate() {
    this.scrollDownHistory();
  }

  scrollDownHistory() {
    const messagesWrapper = this.refs.messagesWrapper;
    if (messagesWrapper) {
      messagesWrapper.scrollTop = messagesWrapper.scrollHeight;
    }
  }

  render() {
    const { classes, messages } = this.props;
    return (
      <div className={classes.messagesWrapper} ref="messagesWrapper">
        {messages && messages.map((message, index) => (
          <ChatMessage key={index} {...message} />
        ))}
        <History />
      </div>
    );
  };
};

export default withStyles(styles)(CHatMessageList);
