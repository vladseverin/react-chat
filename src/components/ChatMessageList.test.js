import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { MemoryRouter, Route } from 'react-router-dom';
import ChatMessageList from './ChatMessageList';

jest.mock('moment', () => () => ({ fromNow: () => 'a few seconds ago' }));

const mockProps = {
  messages: [
    {
      chatId: '123456',
      content: 'Hey, Wassup',
      createdAt: '2018-06-13T20:48:03.200Z',
      sender: {
        firstName: 'Vladislav',
        lastName: 'Severin',
        username: 'vlad',
        _id: '12345',
      },
      _id: '12345',
    },
  ],
  activeUser: {
    firstName: 'Vladislav',
    lastName: 'Severin',
    username: 'vlad',
    isMember: true,
    isCreator: true,
    isChatMember: true,
  },
};

describe('<ChatMessageList />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter initialEntries={['/chat']}>
        <Route
          path="/chat/:chatId?"
          render={props => <ChatMessageList {...mockProps} {...props} />}
        />
      </MemoryRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter initialEntries={['/chat']}>
          <Route
            path="/chat/:chatId?"
            render={props => <ChatMessageList {...mockProps} {...props} />}
          />
              </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
