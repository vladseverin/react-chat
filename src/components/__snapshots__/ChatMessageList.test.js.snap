// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatMessageList /> renders correctly 1`] = `
<div
  className="MuiPaper-root-58 MuiPaper-elevation2-62 MuiPaper-rounded-59 ChatMessageList-paper-56"
>
  <h1
    className="MuiTypography-root-85 MuiTypography-display1-89 MuiTypography-gutterBottom-102"
  >
    Start messagingâ€¦
  </h1>
  <p
    className="MuiTypography-root-85 MuiTypography-body1-94 MuiTypography-gutterBottom-102"
  >
    Use 
    <strong>
      Global
    </strong>
     to explore communities around here.
  </p>
  <p
    className="MuiTypography-root-85 MuiTypography-body1-94 MuiTypography-gutterBottom-102"
  >
    Use 
    <strong>
      Recents
    </strong>
     to see your recent conversations.
  </p>
</div>
`;
